R/tax_unique.R

```diff
@@ -0,0 +1,355 @@
+    }
+  }
+
+  if (!is.null(species)) {
+    #Create genus_species column
+    occurrences$genus_species <- paste(occurrences$genus, occurrences$species)
+    occurrences <- subset(occurrences, select = -c(species))
+
+    #If one or both of the values was NA, set to NA
+    occurrences$genus_species[grep("NA", occurrences$genus_species)] <- NA
+  }
+
+  #Remove absolute repeats
+  occurrences <- unique(occurrences)
+
+  #Create an empty dataset to collect unique occurrences in
+  to_retain <- data.frame()
+
+  if (resolution == "species") {
+    #Retain occurrences identified to species level and remove from dataframe
+    to_retain <- rbind(to_retain, occurrences[!is.na(occurrences$genus_species),
+                                              ])
+    occurrences <- occurrences[is.na(occurrences$genus_species), ]
+
+    #Retain occurrences identified to genus level and not already in dataset
+    to_retain <- rbind(to_retain,
+                       occurrences[(!occurrences$genus %in% c(to_retain$genus,
+                                                              NA)), ])
+    occurrences <- occurrences[is.na(occurrences$genus), ]
+  } else
+
+  if (resolution == "genus") {
+    #Remove genus_species column and remove genus repeats
+    if (!is.null(binomial) || !is.null(names)) {
+    occurrences <- subset(occurrences, select = -c(genus_species))
+    } else
+      if (!is.null(species)) {
+    occurrences <- subset(occurrences, select = -c(genus_species, species))
+      }
+
+    occurrences <- unique(occurrences)
+
+    #Retain genus identifications and remove from dataframe
+    to_retain <- rbind(to_retain, occurrences[!is.na(occurrences$genus), ])
+    occurrences <- occurrences[is.na(occurrences$genus), ]
+  }
+
+  #Retain occurrences identified to family level and not already in dataset
+  to_retain <- rbind(to_retain,
+                     occurrences[!(occurrences$family %in% c(to_retain$family,
+                                                             NA)), ])
+  occurrences <- occurrences[is.na(occurrences$family), ]
+
+  if (!is.null(order)) {
+  #Retain occurrences identified to order level and not already in dataset
+  to_retain <- rbind(to_retain,
+                     occurrences[!(occurrences$order %in% c(to_retain$order,
+                                                            NA)), ])
+  occurrences <- occurrences[is.na(occurrences$order), ]
+  }
+
+  if (!is.null(class)) {
+  #Retain occurrences identified to class level and not already in dataset
+  to_retain <- rbind(to_retain,
+                     occurrences[!(occurrences$class %in% c(to_retain$class,
+                                                            NA)), ])
+  occurrences <- occurrences[is.na(occurrences$class), ]
+  }
+
+  #Produce column with unique taxon names
+  if (resolution == "species") {
+    to_retain$unique_names <- to_retain$genus_species
+  } else {
+    to_retain$unique_names <- NA
+  }
+
+  for (i in 1:nrow(to_retain)) {
```

Comment:
This is probably throwing a lint warning, I had to change this so much in my own code :') I think you need to use something like seq_len(nrow(...))